cmd/libsnap-confine-private/cgroup-support.c

```diff
@@ -66,3 +67,18 @@ void sc_cgroup_create_and_join(const char *parent, const char *name, pid_t pid)
     }
     debug("moved process %ld to cgroup hierarchy %s/%s", (long)pid, parent, name);
 }
+
+static const char *cgroup_dir = "/sys/fs/cgroup";
+// from statfs(2)
+static const int CGROUP2_SUPER_MAGIC =  0x63677270;
+
+bool sc_cgroup_is_v2() {
+    struct statfs buf;
+
+    int err = statfs(cgroup_dir, &buf);
+    if (err == 0 && buf.f_type == CGROUP2_SUPER_MAGIC) {
```

Comment:
I would prefer if we handle `EPERM` and die() (at least for now).

Reply:
Is EPERM just an example? I ask because I don't see it in the statvfs manpage. I guess we can die() on anything but EFAULT or ENOENT?

Reply:
Indeed, perhaps the logic should be swapped. It's okay to have ENOENT but anything else should be die() IMO.

Reply:
Added

Reply:
Nitpick (and sorry for not suggesting this earlier), I think this would be a little easier to read:
```
    if (statfs(cgroup_dir, &buf) != 0) {
        if (errno == ENOENT) {
            return false;
        }
        die("cannot statfs %s", cgroup_dir);
    }
    if (buf.f_type == CGROUP2_SUPER_MAGIC) {
        if (!did_warn) {
            fprintf(stderr, "WARNING: cgroup v2 is not fully supported yet\n");
            did_warn = true;
        }
        return true;
    }
    return false;
```